zproject/backends.py

```diff
@@ -571,7 +571,7 @@ def get_verified_emails(self, *args: Any, **kwargs: Any) -> List[str]:
 
         verified_emails = []  # type: List[str]
         for email_obj in emails:
-            if not email_obj.get("verified"):
+            if not email_obj.get("verified") or email_obj["email"].endswith("noreply.github.com"):
```

Comment:
I would try to avoid this idiom in the future.  Using `continue` inside a `for` loop tends to be brittle.

I would instead extract a tiny function like this:

~~~ py
def eligible_emails(emails):
    return [
        email for email in emails
        if email.obj.get['verified'] and
            not email.endswith("noreply.github.com")
    ]
~~~

It's really easy to write direct tests against that function, and then your loop just becomes:

~~~ py
for email_obj in eligible_emails(emails):
    # start doing real stuff immediately
~~~

Reply:
Thanks @showell , made the required changes.